Author: Matthias Klose <doko@ubuntu.com>
Description: Enable ld --as-needed (basically by using LIBS instead of LDFLAGS.
 Adding libraries in LDFLAGS is almost always a bad idea.
Closes: #607582

--- velvet_1.1.02~nozlibcopy.orig/Makefile
+++ velvet_1.1.02~nozlibcopy/Makefile
@@ -1,14 +1,15 @@
 CC = gcc
 CFLAGS = -Wall
 DEBUG = -g
-LDFLAGS = -lm
+LIBS = -lm
 OPT = -O3
 MAXKMERLENGTH=31
 CATEGORIES=2
 DEF = -D MAXKMERLENGTH=$(MAXKMERLENGTH) -D CATEGORIES=$(CATEGORIES)
 
 Z_LIB_DIR=third-party/zlib-1.2.3
-Z_LIB_FILES=$(Z_LIB_DIR)/*.o
+# In Debian we do not use local zlib copy, however, we need to add $(LIBS) to compile options and thus we tweak this in here:
+Z_LIB_FILES=$(LIBS)
 
 # Mac OS users: uncomment the following lines
 # Z_LIB_FILES=
